source/getting-started/tutorials/timing-manager/index.md

```diff
@@ -0,0 +1,309 @@
+
+To link the ADC to the [Timing Manager](/firmware/arch/timing-manager.md), edit the  `...controller_init()` function to include the function call `timing_manager_enable_sensor(ADC)`:
+
+`app_controller.c`:
+```C
+void app_controller_init(void)
+{
+    // Enable data sampling for ADC
+    timing_manager_enable_sensor(ADC);
+    // Register "ctrl" command with system
+    cmd_controller_register();
+}
+```
+
+### Reading Sensor Data
+Data can be obtained from the sensor interaces in the usual manner, irrespective of the mode of the [Timing Manager](/firmware/arch/timing-manager.md). For example, the ADC can be read via `analog_getf(ANALOG_IN1, &output)` (see the [analog input page](/hardware/subsystems/analog.md)). Note that we won't be using the ADC's actual numerical output in this tutorial, we're just enabling it to activate the [Timing Manager](/firmware/arch/timing-manager.md).
+
+### System Timing
+To understand the specific timings of sensor collection and tasks, we need to know the specific numbers of the factors that control tasks.
+ - User set `TASK_CONTROLLER_UPDATES_PER_SEC` is set in `task_controller.h`, its value is set to `(10000)`
+ - PWM frequency can be set with a hardware command `hw pwm sw`, but the default value is in `common/drv/pwm.h` at `(100000.0)`
+ - User Event Ratio is set in `common/drv/timing_manager.c` with the `timing_manager_init()` function. It is set to `TM_DEFAULT_PWM_RATIO`, which is `10`.
+ - The ADC's data acquisition time can be retrieved with the hardware command `hw tm time adc` or the C function `timing_manager_get_time_per_sensor(ADC)`. It is around 0.86 microseconds.
+ - The control task's Run-Time and Loop-Time can be retrieved with the user-made command `ctrl stats print`. We are specifically looking at the Run-Time.
+
+```
+Task Stats:
+Loop Num:	21431 samples
+Loop Min:	94.52 usec
+Loop Max:	106.21 usec
+Loop Mean:	100.00 usec
+Loop Var:	0.02 usec
+Run Num:	21513 samples
+Run Min:	3.25 usec
+Run Max:	3.92 usec
+Run Mean:	3.40 usec
+Run Var:	0.00 usec
+```
+
+The Loop Mean is how much time there is between successive executions of the control task. It should be `1` / `TASK_CONTROLLER_UPDATES_PER_SEC`. And in this case it is, at `1` / `10000` seconds.
+
+### We can now draw our timing diagram with exact parameters:
+
+![](images/tmPostSensor.svg)
+
+![](images/tmLegacy.svg)
+
+We can see that we are sampling the sensors once per control task. That's because our User Event Ratio of `10` fits perfectly with the ratio between the PWM frequency and our control task's frequency. This is the gold standard. In this tutorial we will experiment with changes to the parameters of the Timing Manager and observe the effects on control task timings.
+
+## Experiment 1 - Ratio is too large
+
+If we increase the User Event Ratio, we can cause the control task to run at less than 10kHz. Let's increase it to 20 by putting `timing_manager_set_ratio(20)` in the `controller_init()` function.
```

Comment:
Change writing perspective a bit to emphasize that this is a mistake. The user is not following the rules set up back at the start of the article (give a link to those rules)